src/hotspot/os/bsd/os_perf_bsd.cpp

```diff
@@ -60,6 +65,34 @@ class CPUPerformanceInterface::CPUPerformance : public CHeapObj<mtInternal> {
   long _total_ticks;
   int  _active_processor_count;
 
+  double normalize(double value) {
+    return MIN2<double>(MAX2<double>(value, 0.0), 1.0);
+  }
+#else
+  struct CPUTicks {
+    uint64_t usedTicks;
+    uint64_t totalTicks;
+  };
+
+  struct JVMTicks {
+    uint64_t userTicks;
+    uint64_t systemTicks;
+    CPUTicks cpuTicks;
+  };
+
+  int _num_procs;
```

Comment:
_active_processor_account may better reflect that on some OSes (IIRC FreeBSD is included) the number of processors and the number of active processors are not necessarily the same.

Reply:
Note that I'm not quite sure how the kern.cp_times sysctl handles this.  Does it return zeroes for inactive processors?  Only records for active processors?  I've found no documentation on that so far and haven't looked through any code yet.

Reply:
Do you want me to hold off on changing the variable name?

Seems like there could be some changes needed if the active processor count can change after initialization. Does FreeBSD support hot enable/disable of cpus?

Reply:
Looking at the code for active_processor_count() on FreeBSD it checks for the process being assigned a certain set of CPUs (per cpuset(1)) rather than hot enable/disable.  That raises a couple of questions here:

* Should we keep CPU totals only for those processors that the JVM is allowed to use?  Given that the set is dynamic, I think the answer is no.
* The return value for kern.cp_times probably doesn't take account of this.  So the array of CPU ticks may be undersized on FreeBSD in this situation and as a result may not even include the CPUs that the JVM is assigned to.

Anyway, after refreshing my memory on what FreeBSD is doing, I think this will require some further investigation and shouldn't block the merge.  We may as well leave the variable name as is for now.

Reply:
Okay, got it. I'll make the other minor requested changes and do a test build before squash merging the branch back into bsd-port.